### PR TITLE
backend-extractor refactoring

### LINKED ISSUE #655: refactor BackendExtractor
Main refactoring was done in #617. But due to deadlines, not all of extractors were refactored.

Please, refer to #617 to see the expected architecture of refactoring the extractor. For code example you can refer to [GroupExtractor](https://github.com/Azure/azure-api-management-devops-resource-kit/blob/main/src/ArmTemplates/Extractor/EntityExtractors/GroupExtractor.cs)

### PR DESCRIPTION
**Pull request by DeagleGross**
Closes #655 

Also removed all unused namespaces using intellisense command

### CODE REVIEW

File: src/ArmTemplates/Extractor/EntityExtractors/BackendExtractor.cs [hunk 1]
```diff
@@ -12,260 +10,214 @@
...
-            {
-                if (namedValueResource.Properties.DisplayName != null && policyContent.Contains(namedValueResource.Properties.DisplayName) ||
-                    namedValueResource.Properties.Value != null && policyContent.Contains(namedValueResource.Properties.Value))
-                {
-                    return true;
-                }
 
-            }
-            return false;
+            return namedValueResourcesUsedByBackend?.Any(x =>
```
[the remark this answers is not in the document]
Comment (DeagleGross) [reply src/ArmTemplates/Extractor/EntityExtractors/BackendExtractor.cs:155]: agree!

File: src/ArmTemplates/Extractor/EntityExtractors/BackendExtractor.cs [hunk 2]
```diff
@@ -12,260 +10,214 @@
...
+                            if (this.DoesPolicyReferenceBackend(policyContent, Array.Empty<NamedValueTemplateResource>(), backendName, backendResource))
                             {
-                                var policyContent = this.policyExtractor.GetCachedPolicyContent(policyTemplateResource, baseFilesGenerationDirectory);
-
-                                if (this.DoesPolicyReferenceBackend(policyContent, emptyNamedValueResources, backendName, backendTemplateResource))
-                                {
-                                    // dont need to go through all policies and backends if the named values has already been found
-                                    return true;
-                                }
+                                // dont need to go through all policies and backends if the named values has already been found
```
[the remark this answers is not in the document]
Comment (DeagleGross) [reply src/ArmTemplates/Extractor/EntityExtractors/BackendExtractor.cs:214]: fixed

File: src/ArmTemplates/Common/Templates/Backend/BackendServiceFabricProperties.cs [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Backend
 {
-    public class BackendSubProperties
+    public class BackendServiceFabricProperties
```
Comment (Hardell) [inline src/ArmTemplates/Common/Templates/Backend/BackendServiceFabricProperties.cs:9]: You have to update the name in the copyright comment above as well.

Comment (Hardell) [reply src/ArmTemplates/Common/Templates/Backend/BackendServiceFabricProperties.cs:9]: Also, I noticed that a bunch of files don't have this header. Do you know if we need it at all? The project is open source. Does Microsoft have copyright even when people from outside of the org contribute?

Comment (DeagleGross) [reply src/ArmTemplates/Common/Templates/Backend/BackendServiceFabricProperties.cs:9]: @Hardell I think we can resolve this for now, and I will create a separate task on removing those comments, if it is required.
What do you think?

Comment (Hardell) [reply src/ArmTemplates/Common/Templates/Backend/BackendServiceFabricProperties.cs:9]: Agreed

**Review by Hardell — APPROVED** [review]

**Pull request merged**